tests/python_package_test/test_basic.py

```diff
@@ -328,3 +328,35 @@ def check_asserts(data):
         lgb_data.set_weight(sequence)
         lgb_data.set_init_score(sequence)
         check_asserts(lgb_data)
+
+    def test_trees_to_dataframe(self):
+
+        def _imptcs_to_numpy(X, impcts_dict):
+            cols = ['Column_' + str(i) for i in range(X.shape[1])]
+            imptcs = []
+            for col in cols:
+                try:
+                    imptcs.append(impcts_dict[col])
+                except KeyError:
+                    imptcs.append(0.)
+            return np.array(imptcs)
+        X, y = load_breast_cancer(True)
+        data = lgb.Dataset(X, label=y)
+        bst = lgb.train({"objective": "binary"}, data, 10)
+        tree_df = bst.trees_to_dataframe()
+        split_dict = (tree_df[~tree_df.split_gain.isnull()]
+                      .groupby('split_feature')
+                      .size()
+                      .to_dict())
+
+        gains_dict = (tree_df
+                      .groupby('split_feature')['split_gain']
+                      .sum()
+                      .to_dict())
+
+        tree_split = _imptcs_to_numpy(X, split_dict)
+        tree_gains = _imptcs_to_numpy(X, gains_dict)
+        mod_split = bst.feature_importance('split')
+        mod_gains = bst.feature_importance('gain')
+        np.testing.assert_equal(tree_split, mod_split)
+        np.testing.assert_allclose(tree_gains, mod_gains)
```

Comment:
Can we have some more tests? For instance, we can check that there are exactly 10 trees in the DataFrame.

Reply:
I added two more tests. One to make sure the node count in the top-level node of each of the trees is the same length as the data and the other to ensure we have 10 trees (which makes me a bit worried as I've seen whole trees get pruned but with `min_gain_to_split` at 0 in this test, we should be fine).

Reply:
Very awesome! Many thanks!